### PR TITLE
Next Depreciation Date is entered as past date

### PR DESCRIPTION
**Pull request by nick9822**
#7722 
Changed validation of next_depreciation date against purchase_date instead of nowdate.

Thanks

### CODE REVIEW

File: erpnext/accounts/doctype/asset/asset.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def validate_asset_values(self):
 			if cint(self.number_of_depreciations_booked) > cint(self.total_number_of_depreciations):
 				frappe.throw(_("Number of Depreciations Booked cannot be greater than Total Number of Depreciations"))
 						
-		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(nowdate()):
+		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(self.purchase_date):
```
Comment (nabinhait) [inline erpnext/accounts/doctype/asset/asset.py]: This is a warning message when user selects the next depreciation date before today. It does not stop to create /submit the asset.

You can add another validation for purchase date, which should block saving the asset.

Comment (nick9822) [inline erpnext/accounts/doctype/asset/asset.py]: Yes I got that warning. But why the next depreciation is validated against nowdate? I guess should be validated against purchase date.

Please advice.

Comment (nabinhait) [inline erpnext/accounts/doctype/asset/asset.py]: If next depreciation date is before nowdate, that means you are creating an existing asset and its depreciation entry is not booked manually via Journal Entry. But we believe for most existing assets, their past depreciation entry should be booked via Journal Entry or in other accounting system (if user migrating to erpnext). There is a huge chance to enter a past date mistakenly. Thats why we added the warning message, maybe we should a "Warning" label before the message.

But we should surely add another validation for purchase date, it definitely can't be before purchase date.

Comment (nick9822) [inline erpnext/accounts/doctype/asset/asset.py]: Thanks for the insight.
Yes you are right, and i guess there should be some indication in asset after saving that their past depreciation entry are not booked.

So i will add "Warning!" text to it and frappe.throw for self.purchase_date as a seperate validation.

**Review by nabinhait — APPROVED** [review]

**Pull request merged**